packages/repository-json-schema/src/build-schema.ts

```diff
@@ -187,20 +238,47 @@ export function modelToJsonSchema(ctor: Function): JSONSchema {
     }
 
     const propSchema = getJsonSchema(referenceType);
+    includeReferencedSchema(referenceType.name, propSchema);
+  }
 
-    if (propSchema && Object.keys(propSchema).length > 0) {
-      result.definitions = result.definitions || {};
+  if (options.includeRelations) {
+    for (const r in meta.relations) {
+      result.properties = result.properties || {};
+      const relMeta = meta.relations[r];
+      const targetType = resolveType(relMeta.target);
+      const targetSchema = getJsonSchema(targetType, options);
+      const targetRef = {$ref: `#/definitions/${targetSchema.title}`};
 
-      // delete nested definition
-      if (propSchema.definitions) {
-        for (const key in propSchema.definitions) {
-          result.definitions[key] = propSchema.definitions[key];
-        }
-        delete propSchema.definitions;
-      }
+      const propDef = relMeta.targetsMany
+        ? <JSONSchema>{
+            type: 'array',
+            items: targetRef,
+          }
+        : targetRef;
 
-      result.definitions[referenceType.name] = propSchema;
+      // IMPORTANT: r !== relMeta.name
+      // E.g. belongsTo sets r="categoryId" but name="category"
```

Comment:
Maybe this is good indication to apply the `belongsTo` decorator to a navigational property like we do with `hasMany` instead of the foreign key? e.g. on `category` instead of `categoryId`

Reply:
I agree with you that `@belongsTo` decorator should not be applied on the foreign key :+1:

IMO, the model should not contain any navigational properties, these properties belong to `ModelWithLink` interface only. I would prefer to apply relational decorators at model-class level.